app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,59 @@
+class PostsController < ApplicationController
+
+  def like
+    @post = Post.all.find(params[:id])
```

Comment:
You don't need `.all` here
Also you can create a `before_action` called e.g. `set_post` excluding actions that don't set post so it would be `before_action :set_post, except: [:new, :index, :create]`
and then add private method:
```
def set_post
  @post = Post.find(params[:id])
end
```

That way you can remove setting post in all methods